### ISSUE TITLE
NTP Top sites/favorites sync and platform parity

### ISSUE THREAD

**Issue by iccub** [brave/brave-ios#6701]

**Comment by anthonypkeane** [comment]
**More info:** https://bravesoftware.slack.com/archives/C02GELCD6BC/p1696121788822279

_The scope of this issue is for V1 - detailed below. Subsequent versions should have a new GH issue._ 

**Design:** https://www.figma.com/file/MUhkPocM62cSGYkZ6CXVpT/Top-Sites---Cross-Platform-Suggestions?type=design&node-id=70%3A14712&mode=design&t=t2sESREz26fPX2cc-1
(Not complete, needs work fr V1)

For v1, display a setting for either Favorites or Top Sites
Under DISPLAY > New Tab Page _(potentially rename to Home Page)_ add the following

BACKGROUND IMAGES
Show Background Image (On/Off)  - Default On
Image Type -> Default Images or Sponsored Images 
_(no change needed here)_

WIDGETS
Privacy Hub (On/Off)  - Default On
Favorites (On/Off)  - Default On
Top Tiles (On/Off)  - Default Off (Greyed out unless Favorites is off)  --> Double check UX on this with Devs and Design

SHORTCUTS
Show Bookmarks Shortcut (On/Off) - Default Off
~~~~~~

**Logic**
Top Sites 
Top sites are automatically generated based on a users browsing history. The logic for displaying these top sites is:

- Frequent Visits: Based on a users browsing history, identify websites they visit most frequently. The more often they visit a particular site, the more likely it is to appear in the Top Sites section.
- Recently Visited: Recent visits to websites also influence the display. Sites a user visited more recently are given priority.
- Private Mode: Top Sites section won't be influenced by sites a user visit in this mode.

In Summary, follow the logic we use on Android for Top Sites
